adding a new APM guide for p99 anomaly monitors on a service (#5193)
DataDog/documentation: content/en/tracing/guide/alert_anomalies_p99_database.md

---
title: Alert on anomalies in database services p99 latency
kind: guide
disable_toc: true
further_reading:
- link: "tracing/setup/"
  tag: "Documentation"
  text: "Learn how to setup APM tracing with your application"
- link: "tracing/visualization/services_list/"
  tag: "Documentation"
  text: "Discover the list of services reporting to Datadog"
- link: "tracing/visualization/resource"
  tag: "Documentation"
  text: "Dive into your resource performance and traces"
- link: "tracing/visualization/trace"
  tag: "Documentation"
  text: "Understand how to read a Datadog Trace"
---

{{< img src="tracing/guide/alert_anomalies_p99_database/alert_anomalies_full.gif" alt="Monitor view with ongoing alert" responsive="true" style="width:90%;">}}

> Datadog allows you to set monitors to keep track of the health of your services with APM instead of constantly monitoring it yourself. In this example we’ll use a special monitor called an anomaly detection monitor. [Anomaly detection][1] is an algorithmic feature that allows you to identify when a metric is behaving differently than it has in the past, taking into account trends, seasonal day-of-week, and time-of-day patterns. It is well-suited for metrics with strong trends and recurring patterns that are hard or impossible to monitor with threshold-based alerting.

Reviewer: ```suggestion
Datadog allows you to set monitors to keep track of the health of your services with APM instead of constantly monitoring it yourself. In this example, we’ll use an anomaly detection monitor. [Anomaly detection][1] is an algorithmic feature that allows you to identify when a metric is behaving differently than it has in the past, taking into account trends, seasonal day-of-week, and time-of-day patterns. It is well-suited for metrics with strong trends and recurring patterns that are hard or impossible to monitor with threshold-based alerting.
```

1. **Open the [New Monitor Page][2] and choose [APM][3]**.
2. **Select the APM Metrics option**.
- **Choose your environment** under Primary Tags.
- **Choose the database to monitor** under Service.
    
    Under Resource, you can choose to monitor specific queries run in the database, but in this example, we’ll look at overall performance so leave it as `*`.

>     Once you choose a service the next step becomes available for you to set and a chart appears at the top of the page showing the performance of the metric the new monitor will track.

Reviewer: ```suggestion
    Once you choose a service, the next step becomes available for you to set, and a chart appears at the top of the page showing the performance of the metric that the new monitor tracks.
```

    {{< img src="tracing/guide/alert_anomalies_p99_database/alert_anomalies_2.png" alt="Monitor view with ongoing alert" responsive="true" style="width:90%;">}}

> 3. **Choose an Anomaly Alert** and under the For option select p99 latency.

Reviewer: ```suggestion
3. **Choose an *Anomaly Alert*** and under the *For* option select p99 latency.
```
we italicize UI elements

    Once you choose Anomaly Alert the chart also shows you the baseline expected behavior for the metric chosen - in our case, p99 latency.

> 4. **Set the “Alert when” field value to 100%**.

Reviewer: ```suggestion
4. **Set the *Alert when* field value to 100%**.
```

    This means that all of the events for the selected duration have to be anomalous for the alert to trigger. This is a best practice for starting with Anomaly Detection. Over time, you’ll find the right values that fit your situation. You can find out more about Anomaly Detection Monitors in the [FAQ][4].

5. **Change the alert notification**. In this example, you can either leave the notification content with the default text or choose team members to tag in the alert.

    {{< img src="tracing/guide/alert_anomalies_p99_database/alert_anomalies_3.png" alt="Monitor view with ongoing alert" responsive="true" style="width:90%;">}}

    You can read more about the markup for notification text and what values and conditions you can set there in the [notifications overview][5].

> 6. **Make sure your username appears in the “Notify your team” box** and add any additional team members that should be notified in case of a database latency anomaly. To add another user be sure to type `@` at the start. **Click on Save**.

Reviewer: ```suggestion
6. **Make sure your username appears in the *Notify your team* box** and add any additional team members that should be notified in case of a database latency anomaly. To add another user be sure to type `@` at the start. **Click on *Save***.
```

    Your alert is now set, you can tweak any of the parameters from this screen and follow the metric performance.

> 7. **Switch from the Edit tab to the Status tab**.

Reviewer: ```suggestion
7. **Switch from the *Edit* tab to the *Status* tab**.
```

    {{< img src="tracing/guide/alert_anomalies_p99_database/alert_anomalies_4.png" alt="Monitor view with ongoing alert" responsive="true" style="width:90%;">}}

>     Here you can see the current status of your monitor, mute it or explore deeper into the specifics of a triggered alert.

Reviewer: ```suggestion
    Here you can see the current status of your monitor, mute it, or explore deeper into the specifics of a triggered alert.
```
